app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -1,14 +1,120 @@
+            "Ser súper estrellas e íntimos amigos tiene sus desventajas, al menos para George. Su esposa, Amal, es muy celosa de Julia e irrumpió varias veces en las grabaciones de su última peli juntos, aunque nunca pescó nada raro.",
+            intArrayOf(1, 2)
+        ),
+        News(
+            8,
+            1,
+            "2016-07-18T14:01:38.673Z",
+            "Hipnosis: la nueva vedette de las neurociencias",
+            "http://bucket1.glanacion.com/anexos/fotos/50/2082050.jpg",
+            "Hace un año, Marisa Bello, bibliotecóloga de La Plata, separada, 51 años, condujo un auto ultralujoso. Luego, durmió profundamente. Más tarde, se rió a carcajadas y olió un perfume indescriptible. Todo eso lo vivió desde una silla, apostada en el escenario de un pabellón de Tecnópolis. Para ella, sucedió durante una hora. \"En realidad, estuvo entre dieciséis y dieciocho minutos, que es el tiempo máximo que utilizamos durante nuestros espectáculos para hipnotizar a la gente -explica Gonzalo Blanc, un abogado de 41 años-. Pero la percepción del tiempo en ese estado es otra, y eso la llevó a sentir la experiencia mucho más larga\". Durante dieciséis presentaciones en Tecnópolis, Gonzalo, junto con el médico Daniel West, de 30 años, practicaron hipnosis colectiva sobre el público. Los dos viven en Montevideo y se dedican desde hace más de diez años a investigar las neurociencias. Dan seminarios, conferencias y talleres empresariales para mejorar el rendimiento a través de la hipnosis -tuvieron clientes como YPF, Telefónica, L'Oréal y Santillana-; practican hipnosis clínica para atenuar el dolor y curar patologías, y sus conferencias en TEDx Durazno y en TEDx Río de la Plata, llamadas \"¿Se puede entrenar a la mente para ser exitosos?\", tienen más de 150.000 reproducciones. ",
+            intArrayOf()
+        ),
+        News(
+            9,
+            2,
+            "2016-07-18T13:53:13.735Z",
+            "Como cuidar los muebles de cuero",
+            "http://bucket3.glanacion.com/anexos/fotos/28/soluciones-2231528w620.jpg",
+            "Generalmente los muebles de cuero son elegidos para destacarse porque resultan muy elegantes. Si bien el material es duradero, requiere de un mayor cuidado y mantenimiento que otros tipos de telas.",
+            intArrayOf(3)
+        )
+    )
+
     override fun layout(): Int = R.layout.fragment_news
 
-    override fun init() {}
+    override fun init() {
+        vNewsSwipeRefreshLayout.setOnRefreshListener {
+            vNewsSwipeRefreshLayout.isRefreshing = false
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        if (newsList.isEmpty()) {
+            Toast.makeText(requireContext(), "Don't exists news to show.", Toast.LENGTH_SHORT).show()
+        } else {
+            super.onViewCreated(view, savedInstanceState)
+            vNewsRecyclerView.apply {
+                layoutManager = LinearLayoutManager(activity)
```

Comment:
the layoutManager definition is ok to be done here, and this code is being run in the init() function, so can go there.

The adapter definition need the newsList, so it should be defined when you already ask for them. Therefore, you are supose to know if the list is empty or not before calling a function to define the NewsAdapter. 

All of this logic is must be done in the presenter, since it is the one who is going to ask for the news to the backend